client/src/partials/patients/record/patient_record.js

```diff
@@ -46,6 +46,13 @@ function PatientRecordController($stateParams, Patients, Notify, moment, Upload,
       vm.patient = result;
       vm.loading = false;
 
+      // get the latest Invoice
+      Patients.latest(vm.patient.debtor_uuid)
+        .then(function (result){
+          vm.patientInvoice = result[0];          
```

Comment:
For the latest invoice, you will always want a single invoice (or none, if the patient was never billed).  Instead of sending an array, you should send a single JSON object to the client.  That way you will not have to do `result[0]`.